### PR TITLE
XDAI Unlock Memberships

### PR DESCRIPTION
**Pull request by blurpesec**
### Description
Sets up xdai unlock memberships

### CODE REVIEW

File: src/features/PurchaseMembership/helpers.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const createApproveTx = (payload: MembershipSimpleTxFormFull): Partial<IT
     baseTokenAmount: toWei(payload.membershipSelected.price, DEFAULT_ASSET_DECIMAL),
     fromAddress: payload.account.address,
     spenderAddress: payload.membershipSelected.contractAddress as TAddress,
-    chainId: DEFAULT_NETWORK_CHAINID,
+    chainId: payload.network.chainId,
```
Comment (FrederikBolding) [inline src/features/PurchaseMembership/helpers.ts:20]: Can we test these helper functions?

File: src/translations/lang/en.json [hunk 2]
```diff
@@ -345,12 +345,14 @@
...
+    "MEMBERSHIP_ACCORDION_FOURTH_TITLE": "Will the price stay the same forever?",
     "MEMBERSHIP_ACCORDION_SECOND_CONTENT": "4 DAI (~$4.00) per month, or you can purchase a 12-month package at a discounted rate. For the hardcore, there’s also a lifetime membership.",
     "MEMBERSHIP_ACCORDION_SECOND_TITLE": "How much does the MyCrypto Membership cost?",
     "MEMBERSHIP_ACCORDION_THIRD_CONTENT": "We use the [Unlock Protocol](https://unlock-protocol.com/) for this. You’ll be purchasing a token that unlocks features of MyCrypto when we detect that you hold that token. Each token has an expiration date according to the duration of membership you purchased.",
     "MEMBERSHIP_ACCORDION_THIRD_TITLE": "How does the MyCrypto Membership work?",
+    "MEMBERSHIP_ACCORDION_FIFTH_CONTENT": "You can purchase a MyCrypto Membership with DAI and ETH, or on the xDai network with [xDai](https://support.mycrypto.com/general-knowledge/scaling/how-to-get-xdai). We will likely be adding support for more networks and tokens in the future.",
```
Comment (FrederikBolding) [inline src/translations/lang/en.json:354]: This line reads wrong to me. Might just be me though 😆

File: src/features/PurchaseMembership/config.ts [hunk 3]
```diff
@@ -133,9 +182,10 @@ export const MEMBERSHIP_CONTRACTS = Object.fromEntries(
   ])
 );
 
-export const MEMBERSHIP_CONTRACTS_ADDRESSES = Object.values(MEMBERSHIP_CONFIG).map(
-  (membership) => membership.contractAddress
-);
+export const getMembershipContracts = (membershipNetworkId: NetworkId) =>
```
Comment (FrederikBolding) [inline src/features/PurchaseMembership/config.ts:185]: Can we move this to a helper file and test?

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/services/Store/store/membership.slice.ts [hunk 4]
```diff
@@ -87,14 +93,33 @@ export function* fetchMembershipsSaga() {
...
-    const memberships = yield call(MembershipApi.getMemberships, relevantAccounts, network);
-    yield put(slice.actions.setMemberships(memberships));
+    const ethereumMemberships = yield call(
+      MembershipApi.getMemberships,
+      ethereumAccounts.map(({ address }) => address as TAddress),
+      ethereumNetwork
+    );
+    const xdaiMemberships = yield call(
+      MembershipApi.getMemberships,
+      xdaiAccounts.map(({ address }) => address as TAddress),
```
Comment (FrederikBolding) [inline src/services/Store/store/membership.slice.ts:119]: Can we add some XDAI accounts to the test to make sure that everything works?

Comment (blurpesec) [reply src/services/Store/store/membership.slice.ts:119]: I don't think that's really necessary as we're mocking the `getMemberships` call/results anyways.

File: src/translations/lang/en.json (hunk 2 above)
Comment (blurpesec) [reply src/translations/lang/en.json:354]: this is jordan-submitted copy. bring it up with the bossman xD

File: src/services/Store/store/membership.slice.ts (hunk 4 above)
Comment (FrederikBolding) [reply src/services/Store/store/membership.slice.ts:119]: Yes, but this means that the xDAI code could not work and we wouldn't spot it. So I would prefer if it ran 😄

File: src/translations/lang/en.json (hunk 2 above)
Comment (blurpesec) [reply src/translations/lang/en.json]: Updated as per discussion w/ bossmang

**Review by FrederikBolding — APPROVED** [review]
lgtm

**Pull request merged**